src/components/AdditionColumn/AdditionColumn.tsx

```diff
@@ -0,0 +1,180 @@
+import { FormEvent, useEffect, useRef, useState } from "react";
+import styled from "styled-components";
+import { Add, Close } from "@material-ui/icons";
+
+import { ProjectContext, ProjectReducerContext } from "../../context/project";
+import { addedNewColumn } from "../../store/project";
+import { CREATE_COLUMN } from "../../api/columns";
+import { useMutation } from "../../hooks/useMutation";
+import { useTypeSafeContext } from "../../hooks/useTypeSafeContext";
+
+import Column, { ColumnResponse } from "../../interfaces/Column";
+
+import Button from "../Button";
+import Card from "../Card";
+import LoadingSpinner from "../LoadingSpinner";
+
+export default function AdditionColumn(): JSX.Element {
+  const project = useTypeSafeContext(ProjectContext);
+  const dispatch = useTypeSafeContext(ProjectReducerContext);
+
+  const editableTitleRef = useRef<HTMLInputElement>(null);
+  const { call: requestCreateColumn, isLoading } = useMutation<ColumnResponse>(CREATE_COLUMN(project.id), "POST");
+
+  const [isColumnExpaned, setIsColumnExpanded] = useState(false);
+  const [columnTitleValue, setColumnTitleValue] = useState("");
+
+  // It should focus on text input when the component is expanded
+  useEffect(() => {
+    editableTitleRef?.current && editableTitleRef.current.focus();
+  }, [isColumnExpaned]);
+
+  const handleEditableTitleChange = (e: FormEvent<HTMLInputElement>) => {
+    setColumnTitleValue(e.currentTarget.value);
+  };
+
+  const handleOnClickAddColumn = () => {
+    setIsColumnExpanded(true);
+  };
+
+  const handleOnClineCloseButton = () => {
+    if (isLoading) return;
+    setIsColumnExpanded(false);
+  };
+
+  const handleOnClickAddButton = async () => {
+    if (isLoading) return;
+    const newColumn = await requestCreateColumn({ variables: { name: columnTitleValue } });
+
+    if (newColumn) {
+      const mappedNewColumn: Column = {
+        id: newColumn.id,
+        name: newColumn.name,
+        order: newColumn.sort,
+        tickets: [],
+      };
```

Comment:
These lines probably can go under `api` or `util` as we might see this pattern often